src/components/comments/comments.service.js

```diff
@@ -0,0 +1,30 @@
+export class CommentsService {
+  constructor($q, $http) {
+    'ngInject;'
+    this.$q = $q;
+    this.$http = $http;
+    this.baseUrl = '/comments/';
+  }
+  
+  getComments(albumId) {
+    let queryUrl =  this.baseUrl + albumId;
+    let deferred = this.$q.defer();
+
+    this.$http.get(queryUrl)
+      .success((response) => deferred.resolve(response))
+      .error((response) => deferred.reject(response));
+
+    return deferred.promise;
+  }
+
+  putComment(comment) {
+    let queryUrl = this.baseUrl;
+    let deferred = this.$q.defer();
+
+    this.$http.post(queryUrl, comment)
+      .success((response) => deferred.resolve(response))
```

Comment:
use `.then()` because `.success` has been deprecated and is causing the application fail.
![image](https://cloud.githubusercontent.com/assets/3399236/21505328/03f84f60-cc45-11e6-9593-b2760d65cf2a.png)